lib/logstash/outputs/s3.rb

```diff
@@ -183,7 +185,7 @@ def register
       raise LogStash::ConfigurationError, "Logstash must have the permissions to write to the temporary directory: #{@temporary_directory}"
     end
 
-    if @validate_credentials_on_root_bucket && !WriteBucketPermissionValidator.valid?(bucket_resource)
+    if @validate_credentials_on_root_bucket && !WriteBucketPermissionValidator.new(@logger).valid?(bucket_resource)
```

Comment:
The logstash-core-plugin-api dependency was left at >= 1.60 for 4.0 meaning 2.4.x stacks, such as mine, could install it. That should have been bumped if it was only 5.x onwards compat! There also seems to be no way to target/install a specific version, so stuck with this broken version till either logstash-core-plugin-api bumped up or 2.4.x backwards compat introduced.